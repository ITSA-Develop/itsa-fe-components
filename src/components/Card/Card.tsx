import { Card as AntCard, CardProps } from 'antd';

export const Card = ({ ...rest }: CardProps) => {
	return <AntCard {...rest} />;
};
