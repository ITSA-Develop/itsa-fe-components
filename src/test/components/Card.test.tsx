import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Card } from '../../components/Card/Card';

describe('Card component', () => {
	it('renders children content', () => {
		const { container } = render(<Card>Test content</Card>);
		expect(screen.getByText('Test content')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('accepts and applies title prop', () => {
		const { container } = render(<Card title="Card title">Content</Card>);
		expect(screen.getByText('Card title')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('accepts extra prop', () => {
		const { container } = render(
			<Card title="Title" extra={<button>Extra</button>}>
				Content
			</Card>,
		);
		expect(screen.getByText('Extra')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders without crashing with no props', () => {
		const { container } = render(<Card />);
		expect(container).toMatchSnapshot();
	});
});
