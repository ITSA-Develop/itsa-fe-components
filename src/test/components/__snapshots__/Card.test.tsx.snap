// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > accepts and applies title prop 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-l9pxc0"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Card title
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      Content
    </div>
  </div>
</div>
`;

exports[`Card component > accepts extra prop 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-l9pxc0"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Title
        </div>
        <div
          class="ant-card-extra"
        >
          <button>
            Extra
          </button>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      Content
    </div>
  </div>
</div>
`;

exports[`Card component > renders children content 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-l9pxc0"
  >
    <div
      class="ant-card-body"
    >
      Test content
    </div>
  </div>
</div>
`;

exports[`Card component > renders without crashing with no props 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-l9pxc0"
  >
    <div
      class="ant-card-body"
    />
  </div>
</div>
`;
